### PR TITLE
DIFM Lite: Initialise the 4th page from the site category

### PR DESCRIPTION
**Pull request by aneeshd16**
#### Changes proposed in this Pull Request

This PR will be merged into the parent PR: https://github.com/Automattic/wp-calypso/pull/59895.

* Initializes the 4th-page section for the Website Content step from the site category selected in the Design Picker step. This mimics the behaviour of the current form.

| Design Category | Title of the 4th page |
|----|----|
|Professional Services | Services |
|Local Services | Services |
|Creative Arts | Portfolio |
|Restaurant | Menu |
|Blog | Blog |

#### Testing instructions

* Purchase DIFM Lite by proceeding through the flow at `/start/do-it-for-me`.
* In the design picker step, note down the category selected.
* After checkout, on the Website Content step, check the title of the 4th-page section. Confirm that it matches the title from the table above.

### CODE REVIEW

**Comment by dzver** [comment]
I'm not sure what I do wrong but I can't seem to test the flow. When I select a theme, I don't go to the next step. I see no JS errors. Do you have any suggestions on what could be wrong?

https://user-images.githubusercontent.com/82778/153573971-619b87f1-1033-4a02-8b0e-f8e27a35707b.mov

public-api is sandboxed, and the sandbox is updated.

**Comment by aneeshd16** [comment]
> I'm not sure what I do wrong but I can't seem to test the flow. When I select a theme, I don't go to the next step. I see no JS errors. Do you have any suggestions on what could be wrong?

Thanks for the review! You just caught a bug with the sign-up flow, which was not related to this PR. :)

The issue happened because the `forceAutoGeneratedBlogName` dependency was not set if the user does not have any existing sites. This caused new site creation to fail with the error "blog_name was not set". I have fixed this in ca3d0c5a3bc05ace7f0107fe266b187bec71a350. Can you please test this again?

File: client/state/selectors/is-difm-lite-website-content-submitted.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import type { AppState, SiteId } from 'calypso/types';
+
+/**
+ * Returns if the DIFM Lite Website Content form has been submitted.
+ *
+ * @param  {object}   state  Global state tree
+ * @param  {number}   siteId Site ID
+ * @returns {boolean}        true if submitted, false otherwise
+ */
+export default function isDIFMLiteWebsiteContentSubmitted(
```
Comment (aneeshd16) [inline client/state/selectors/is-difm-lite-website-content-submitted.ts:12]: This function is not used in this change, but will be used for completing the task "Block access to the form if the purchase is not valid or if the form has already been submitted" (see #59895)

File: client/lib/signup/step-actions/index.js [hunk 2]
```diff
@@ -1256,6 +1256,7 @@ export function isNewOrExistingSiteFulfilled( stepName, defaultDependencies, nex
 		if ( stepProvidesDependency ) {
 			dependency = {
 				newOrExistingSiteChoice: 'new-site',
+				forceAutoGeneratedBlogName: true,
```
Comment (aneeshd16) [inline client/lib/signup/step-actions/index.js:1259]: Reference for this change: https://github.com/Automattic/wp-calypso/pull/60998#issuecomment-1036099100

File: client/state/signup/steps/website-content/reducer.ts [hunk 3]
```diff
@@ -20,16 +20,15 @@ export default withSchemaValidation(
...
 				};
 			case SIGNUP_STEPS_WEBSITE_CONTENT_INITIALIZE_PAGES: {
-				// Initialize with provided pages only if given pages do not exist
 				// When initializing page we need to leave pages that already exist alone
 				// so that persisted state can load correctly
-				const existingPageIDs = state.websiteContent.map( ( page ) => page.id );
-				const newPages = action.payload.filter(
-					( page: PageData ) => ! existingPageIDs.includes( page.id )
-				);
+				const newPages = action.payload.map( ( page: PageData ) => ( {
```
Comment (aneeshd16) [inline client/state/signup/steps/website-content/reducer.ts:25]: This change fixes this bug:
1. Go to `/start/site-content-collection/website-content?siteSlug=<slug1>`. Confirm that there are 4 pages.
2. Go to `/start/site-content-collection/website-content?siteSlug=<slug2>`. `slug2` should be a website with a different category. For example, if `slug1` is "local-services", then `slug2` can be "restaurant". On visiting this URL, 5 pages are visible.

The number of pages shown should be fixed and should always be initialized [here](https://github.com/Automattic/wp-calypso/pull/60998/files#diff-16e080d2ad3a8e19863abe08ffd798f9be323293b0171c0e6f5efdaa7dedd59fR68).

Comment (ddc22) [reply client/state/signup/steps/website-content/reducer.ts:25]: Nice catch! :)

**Review by ddc22 — APPROVED** [review]
LGTM 🚢

**Pull request merged**